src/EmailValidator.java

```diff
@@ -0,0 +1,18 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class EmailValidator {
+    /**
+     * @param str which should be checked;
+     * @return true - validate, false - not;
+     */
+    public boolean validate(String str) {
+        Pattern email = Pattern.compile("([A-z0-9_-]+\\.)*[A-z0-9_-]+@[a-z0-9_-]+(\\.[a-z0-9_-]+)*\\.[a-z]{2,6}");
+        Matcher checkEmail = email.matcher(str);
+        if (!checkEmail.matches()) {
+            throw new UserIncorrectException("Incorrect input");
```

Comment:
1. It should be `IncorrectInputException`
2. Exceptions **always**  should have error context. Why exception is happend? For example it can be  

IncorrectInputException("Email {INPUT} is incorect. Valid pattern is {PATTERN}"). Use https://www.javatpoint.com/java-string-format API for message generation